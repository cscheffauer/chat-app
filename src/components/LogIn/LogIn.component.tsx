import React, { FormEvent, useState } from 'react'
import './LogIn.scss';

interface Props {
    logInUser: (event: any) => void
}

const LogIn = ({ logInUser }: Props) => {
    const [username, setusername] = useState("");

    const changeUsername = (event: FormEvent<HTMLInputElement>) => {
        event.preventDefault();
        setusername(event.currentTarget.value);
    }

    const handleSubmit = (event: FormEvent) => {
        event.preventDefault();
        logInUser(username)
    }
    return (
        <div className="login">
            <p>Please type in your name to join:</p>
            <form onSubmit={handleSubmit}>
                <input onChange={changeUsername}></input>
                <button type="submit">Join</button>
            </form>
        </div>
    )
}

export default LogIn;
