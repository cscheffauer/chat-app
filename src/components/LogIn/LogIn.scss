.login {
	text-align: center;
	margin-top: 5vh; /* Fallback for browsers that do not support Custom Properties */
	margin-top: calc(var(--vh, 1vh) * 5);
}
