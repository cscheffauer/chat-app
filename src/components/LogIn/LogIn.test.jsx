import React from 'react';
import { mount } from 'enzyme';
import LogIn from './LogIn.component';

describe('<LogIn />', () => {
	let wrapper;
	const logInUserMock = jest.fn();

	beforeEach(() => {
		wrapper = mount(<LogIn logInUser={logInUserMock} />);
	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	it('expect to render LogIn component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('Testing the Login Form', () => {
		it('expect to call LogIn fn with typed in username', () => {
			const usernameInput = wrapper.find('input');
			usernameInput.instance().value = 'Christoph';
			usernameInput.simulate('change');

			wrapper
				.find('form')
				.props()
				.onSubmit({ preventDefault() {} }); //submit form
			expect(logInUserMock).toHaveBeenCalledWith('Christoph'); //Christoph should have been passed to loginUserMock
		});

		it('expect to not call LogIn fn with nothing typed in', () => {
			const usernameInput = wrapper.find('input');
			usernameInput.instance().value = '';
			usernameInput.simulate('change');

			wrapper
				.find('form')
				.props()
				.onSubmit({ preventDefault() {} }); //submit form
			expect(logInUserMock).toHaveBeenCalledTimes(0); //logInUserMock should not have been called
		});
	});
});
