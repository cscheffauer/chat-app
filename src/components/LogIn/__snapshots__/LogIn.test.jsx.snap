// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> expect to render LogIn component 1`] = `
<LogIn
  logInUser={[MockFunction]}
>
  <div
    className="login"
  >
    <p>
      Please type in your name to join:
    </p>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
      />
      <button
        type="submit"
      >
        Join
      </button>
    </form>
  </div>
</LogIn>
`;
